src/Microsoft.Azure.WebJobs.Host/Hosting/WebJobsBuilderExtensions.cs

```diff
@@ -164,7 +168,8 @@ public static IWebJobsBuilder UseExternalStartup(this IWebJobsBuilder builder, I
         /// <returns>The updated <see cref="IHostBuilder"/> instance.</returns>
         public static IWebJobsBuilder UseExternalStartup(this IWebJobsBuilder builder, IWebJobsStartupTypeLocator startupTypeLocator, ILoggerFactory loggerFactory)
         {
-            Type[] types = startupTypeLocator.GetStartupTypes();
+            IEnumerable<Type> types = startupTypeLocator.GetStartupTypes()
+                .Where(t => typeof(IWebJobsConfigurationStartup).IsAssignableFrom(t));
```

Comment:
No, this is wrong. I've got it fixed -- this is meant to use `IWebJobsStartup` while UseExternalConfigurationStartup uses `IWebJobsConfigurationStartup`. 

I also figured out why the tests passed -- I had added `IWebJobsConfigurationStartup` to the existing test implementation, so this all worked (since it implemented both interfaces). I've split it out now to have one that only does `IWebJobsStartup` and the tests failed until I fixed it.